### DISCUSSION TITLE
Announcement about the homepage

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#6894] [Q&A] [answered]
I noticed that in the official website document of material, there is no announcement on the homepage, but there is on other pages. How is this achieved?

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/ae813ea1-44e7-4c93-ae6f-cfca1377ef8f)
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/5ff94ce3-0267-42fc-99dd-7897795d94b4)

I've looked at `docs/index.md` and `src/overrides/home.html` but I don't find any information

**Comment by w1ndys** [comment]
I seem to have discovered the key point, is it here?
https://github.com/squidfunk/mkdocs-material/blob/fdeafbb875fe7cf340da429c90e8eaa965e9663e/src/overrides/home.html#L32C1-L59C6

**Reply by kamilkrzyskow** [reply]
The front page (https://squidfunk.github.io/mkdocs-material/) source code is only visible on the Insiders branch, limited to sponsors. 
https://github.com/squidfunk/mkdocs-material-insiders/tree/master/src/overrides
The code you found is the community version of the page which is visible when you `mkdocs serve` the files yourself.
![image](https://github.com/squidfunk/mkdocs-material/assets/34622465/4db116af-86fe-4f90-95e0-46da12b8119d)

**Comment by w1ndys** [comment]
I seem to have found some settings
https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/?h=template#overriding-templates
https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template

Ahahaha, I thought this was a sponsored feature at first

**Reply by kamilkrzyskow** [reply]
The `template: custom.html` is a feature of MkDocs itself and not related to the theme. As for the blog template, there are further iterations of it in Insiders: https://squidfunk.github.io/mkdocs-material/insiders/#goals but the basic version is available in the community version, since a while now.

**Comment by w1ndys** [comment]
But I read the document for a long time and didn't learn how to replace index.html. I just did the following, but the homepage still had no effect.

this is  a  test html
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/25a8fa17-3a0d-42fe-b9a3-8e916edcbd01)

And this index.md
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/f34ff182-8e3f-462b-9413-86b54e0d17e6)
But nothing changes on my homepage

**Reply by w1ndys** [reply]
Oh oh oh oh oh oh, I have implemented the replacement of HTML, but I still haven’t learned how to replace this template based on the variables of material. In other words, I don’t know how to replace the homepage to a state that is fully integrated with material. , I would just replace a complete HTML hardcoded file
If you have time, thank you very much for providing me with some information

**Reply by kamilkrzyskow** [reply]
Like I explained in:
- https://github.com/squidfunk/mkdocs-material/discussions/6888#discussioncomment-8729715

the Jinja template is divided into "blocks" which are predefined in the `base.html` file, and `main.html` (the entry point) **extends** the `base.html` allowing to override and replace the blocks.

When you set `template: home.html` you should extend `main.html` to have access to your default overrides of `base.html` and then override them again for the `announce` block: `{% block announce %}{% endblock %}`. Just leave it empty to "remove it" on the home page.

**Comment by w1ndys** [comment]
@kamilkrzyskow 
Thank you for your reply. I have read all your replies. Another question is that if I not only want to close the announcement on the homepage, I also want to set the homepage as a custom page, just like the start page of material. Is this a sponsored exclusive feature? If not, what should I do?

**Reply by kamilkrzyskow** [reply, marked as answer]
> Another question is that if I not only want to close the announcement on the homepage, I also want to set the homepage as a custom page, just like the start page of material.

I don't understand. You can set the `template: home.html` in any `.md` file, `index.md` or `special-page.md`, and it will be rendered according to the template you provided. So the template without the announce block can be used on the homepage and custom page.

Recently in https://github.com/squidfunk/mkdocs-material/discussions/6853 a user provided his docs and they also use a custom home.html, maybe it will be sufficient as reference:
- https://github.com/mer0x/merox.cloud/blob/master/assets/home.html

If not provide more details about your use case. How is the homepage different from the custom page?

**Reply by w1ndys** [reply]
Thank you, this is very helpful to me.

**Comment by condor-the-bird** [comment]
I still don't see an easy way to configure a landing. Really. I managed everything: snippets, sorting tables, tabs, but this escapes me. I don't get it, sorry. Doesn't work.

**Reply by condor-the-bird** [reply]
Ps: it would be cool to have a generic example, let it be a blank page with placeholders for the top menu.

**Reply by A3Bagged** [reply]
A home.html pretty much is a blank page. Anything inside `{% block tabs %}` can be used as homepage, you can see it as a second content container. Pretty much like:
```html
{#-
  This file was automatically generated - do not edit
-#}
{% extends "main.html" %}
{% block styles %}
  {{ super() }}
  <link rel="stylesheet" href="{{ 'assets/stylesheets/home.css' | url }}">
  <!-- load here any .css files you want to use for the homepage, these only get loaded on this page -->
{% endblock %}
{% block tabs %}
  {{ super() }}
  <!-- Put here your new homepage -->
{% endblock %}
```

Where it says `<!-- Put here your new homepage -->` you can pretty much start off with a new container, that's what i did.

```html
{% block tabs %}
  {{ super() }}
  <style>.md-header{position:initial}.md-main__inner{margin:0}.md-content{display:none}@media screen and (min-width:60em){.md-sidebar--secondary{display:none}}@media screen and (min-width:76.25em){.md-sidebar--primary{display:none}}</style>
  <section class="lp-container">
      {% include "partials/hero.html" %}
      {% include "partials/features.html" %}
      {% include "partials/components.html" %}
      {% include "partials/faq.html" %}
      {% include "partials/calltoaction.html" %}
  </section>
{% endblock %}
```

To split up things in multiple html files so you don't have one long code, you can include them using `{% include "filename.html" %}` i think these are part of [Django template language](https://docs.djangoproject.com/en/5.1/ref/templates/language/). This is not necessary but it will help if you ever have to look through your code to split it up in sections, plus you can enable/disable parts of your homepage by just commenting out one section.

**Reply by squidfunk** [reply]
@condor-the-bird Please understand that we're currently focusing on [foundational work](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/). This also includes making customization of the landing page easier, and in general, to provide better templates that can be configured/customized in minutes.

We understand that the situation is currently not ideal and requires too much web dev knowledge, and we're working on it.
